### ISSUE TITLE
sunodo create should fail when --branch doesn't exists

### ISSUE THREAD

**Issue by endersonmaia** [sunodo/sunodo#186]
## 🙂 Expected behavior

When calling `sunodo create --branch=wrong-branch-name`, the CLI should show an error informing the branch doesn't exists.

## 🫠 Actual behavior

The CLI show that the application was successfully created, instead.

## 🧪 Minimal test case

```
$ sunodo create --template=javascript --branch=wrong-branch-name my-js-dapp
✔ Application created at /home/endersonmaia/Development/my-js-dapp
$ ls -1 my-js-dapp/
Dockerfile
package.json
README.md
src
yarn.lock
```

## 🌎 Environment

What is the environment that the bug happens?

I'm using sunodo CLI 0.7.0

**Comment by DiegoSena** [comment]
@tuler All braches on sunodo-templates are eligible or do we have a fixed list ?

**Comment by endersonmaia** [comment]
@DiegoSena we lock the branch used based on sunodo/sdk version.

For ex., for @sunodo/cli:0.10.4, we use the branch `sdk-0.2` as you can see here https://github.com/sunodo/sunodo/blob/%40sunodo/cli%400.10.4/apps/cli/src/commands/create.ts#L7

Is that what you asked?

**Comment by DiegoSena** [comment]
Not exactly. We can use branch flag on CLI and can literally use any branch on sunodo-template repo. We should allow select ANY BRANCH or based on sunodo version we can select only a few listed branches?

**Comment by tuler** [comment]
I think that flag is more targeted to developers who want to test new templates. Regular users should use just the “main” branch configured for the CLI version.

Maybe the flag could be hidden?

**Comment by endersonmaia** [comment]
That and there's an idea to make it possible to define a different repository also.

In that case, it makes more sense for regular users. Maybe that could go on another issue.
